### ISSUE TITLE
There are some conflicts configuration format pattern includes dot character in name of environment variable

### ISSUE THREAD

**Issue by muratsplat** [erlio/docker-vernemq#61]
Hey Guys
It seems  follow issue is not resolved. https://github.com/erlio/docker-vernemq/issues/14

```env
    - name: "DOCKER_VERNEMQ_VMQ_WEBHOOKS__POOL_timeout"
      value: "60000"
```

I was able to set  as  above  on kubernetes yaml file..

`DOCKER_VERNEMQ_VMQ_WEBHOOKS.POOL_TIMEOUT` is not parsed/resolved by vernemq..

**Comment by cbluth** [comment]
Hi @muratsplat 

I have a working set with a dot in the variable name, see here:

```
cbluth@dl7490:~/x$ kubectl -n platform get statefulset vernemq -oyaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  annotations:
  creationTimestamp: 2018-07-17T10:06:59Z
  generation: 41
  labels:
    app: vernemq
  name: vernemq
  namespace: platform
  resourceVersion: "12909127"
  selfLink: /apis/apps/v1/namespaces/platform/statefulsets/vernemq
  uid: 249195ba-89a9-11e8-bb33-525400ac45ea
spec:
  podManagementPolicy: OrderedReady
  replicas: 4
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: vernemq
  serviceName: vernemq
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: vernemq
    spec:
      containers:
      - env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: DOCKER_VERNEMQ_DISCOVERY_KUBERNETES
          value: "1"
        - name: DOCKER_VERNEMQ_MAX_ONLINE_MESSAGES
          value: "-1"
        - name: DOCKER_VERNEMQ_ERLANG.async_threads
          value: "1024"
        - name: DOCKER_VERNEMQ_KUBERNETES_APP_LABEL
          value: vernemq
        - name: DOCKER_VERNEMQ_KUBERNETES_INSECURE
          value: "1"
        - name: DOCKER_VERNEMQ_KUBERNETES_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: DOCKER_VERNEMQ_ERLANG__DISTRIBUTION__PORT_RANGE__MINIMUM
          value: "9100"
        - name: DOCKER_VERNEMQ_ERLANG__DISTRIBUTION__PORT_RANGE__MAXIMUM
          value: "9109"
        - name: DOCKER_VERNEMQ_LISTENER__VMQ__CLUSTERING
          value: 0.0.0.0:44053
        - name: DOCKER_VERNEMQ_VMQ_PASSWD__PASSWORD_FILE
          value: /etc/vernemq-passwd/vmq.passwd
        image: erlio/docker-vernemq:latest
        imagePullPolicy: IfNotPresent
        lifecycle:
          preStop:
            exec:
              command:
              - /bin/bash
              - -c
              - /usr/sbin/vmq-admin cluster leave node=VerneMQ@${MY_POD_NAME}.vernemq.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.cluster.local
                ; sleep 1 ; /usr/sbin/vmq-admin cluster leave node=VerneMQ@${MY_POD_NAME}.vernemq.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.cluster.local
                -k
        name: vernemq
        ports:
        - containerPort: 1883
          name: mqtt
          protocol: TCP
        - containerPort: 4369
          name: epmd
          protocol: TCP
        - containerPort: 44053
          name: vmq
          protocol: TCP
        - containerPort: 9100
          protocol: TCP
        - containerPort: 9101
          protocol: TCP
        - containerPort: 9102
          protocol: TCP
        - containerPort: 9103
          protocol: TCP
        - containerPort: 9104
          protocol: TCP
        - containerPort: 9105
          protocol: TCP
        - containerPort: 9106
          protocol: TCP
        - containerPort: 9107
          protocol: TCP
        - containerPort: 9108
          protocol: TCP
        - containerPort: 9109
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/vernemq-passwd
          name: vernemq-passwd
          readOnly: true
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - name: vernemq-passwd
        secret:
          defaultMode: 420
          secretName: vernemq-passwd
  updateStrategy:
    type: OnDelete
status:
  collisionCount: 0
  currentRevision: vernemq-6bb75f9b4b
  observedGeneration: 41
  readyReplicas: 4
  replicas: 4
  updateRevision: vernemq-6484c49d5f
  updatedReplicas: 4
cbluth@dl7490:~/x$ 
```


Note the lower-case variable:
```
        - name: DOCKER_VERNEMQ_ERLANG.async_threads
          value: "1024"
```

now, when i check inside the pod:
```
cbluth@dl7490:~/x$ kubectl exec -n platform vernemq-0 -- bash -c 'env | grep async_threads ; echo ; grep async_threads /etc/vernemq/vernemq.conf'
DOCKER_VERNEMQ_ERLANG.async_threads=1024

erlang.async_threads = 64
erlang.async_threads=1024
cbluth@dl7490:~/x$ 
```

^^^ this behaviour is due to a bug in the entrypoint script that reads the env vars

**Comment by muratsplat** [comment]
@cbluth  Thank you for replay 
I think that this point should explained in official documentation. There are a couple exceptions which need to be clear in here..

**Comment by larshesel** [comment]
Hi, I don't really understand what this is about, but we'd be happy to accept a PR to improve the docs!

**Comment by muratsplat** [comment]
Hi @larshesel 

There is a problem on some conf params which has dots and after underscore characters.

For example:
```
// vernemq.conf
vmq_webhooks.pool_timeout = 60000
```
expected format in Environment for above param:  `DOCKER_VERNEMQ_VMQ_WEBHOOKS__POOL_TIMEOUT=60000` but this format is not resolved by the app. Unlike `DOCKER_VERNEMQ_VMQ_WEBHOOKS__POOL_timeout=60000` format is resolved instead of `vmq_webhooks.pool_timeout` in `vernemq.conf`.

In result there is a conflict about format of configuration.

**Comment by cbluth** [comment]
@muratsplat 
create an environment variable like so:
```
        - name: DOCKER_VERNEMQ_VMQ_WEBHOOKS.pool_timeout
          value: "60000"
```
this will workaround the bug and create your desired config.

**Comment by muratsplat** [comment]
Thank you for supporting..

**Comment by larshesel** [comment]
Merged #62
